src/crnlib.rs

```diff
@@ -0,0 +1,144 @@
+use crate::crunch::crn_decomp::CrnHeader;
+
+// Supported compressed pixel formats.
+// Basically all the standard DX9 formats, with some swizzled DXT5 formats
+// (most of them supported by ATI's Compressonator), along with some ATI/X360 GPU specific formats.
+#[derive(PartialEq, PartialOrd)]
+#[repr(C)]
+pub enum CrnFormat {
+    CCrnfmtInvalid = -1,
+
+    CCrnfmtDxt1 = 0,
+
+    // cCRNFmtFirstValid = crn_format::cCRNFmtDXT1 as isize, // Rust doesn't allow same value enums, and as far as I see this is not used in the lib.
+
+    // cCRNFmtDXT3 is not currently supported when writing to CRN - only DDS.
+    CCrnfmtDxt3,
+
+    CCrnfmtDxt5,
+
+    // Various DXT5 derivatives
+    CCrnfmtDxt5CcxY,    // Luma-chroma
+    CCrnfmtDxt5XGxR,    // Swizzled 2-component
+    CCrnfmtDxt5XGbr,    // Swizzled 3-component
+    CCrnfmtDxt5Agbr,    // Swizzled 4-component
+
+    // ATI 3DC and X360 DXN
+    CCrnfmtDxnXy,
+    CCrnfmtDxnYx,
+
+    // DXT5 alpha blocks only
+    CCrnfmtDxt5a,
+
+    CCrnfmtEtc1,
+    CCrnfmtEtc2,
+    CCrnfmtEtc2a,
+    CCrnfmtEtc1s,
+    CCrnfmtEtc2as,
+
+    CCrnfmtTotal,
+
+    CCrnfmtForceDword = 0xFFFFFFFF
+}
+
+#[repr(C)]
+pub struct CrnTextureInfo {
+    pub m_struct_size: u32,
+    pub m_width: u32,
+    pub m_height: u32,
+    pub m_levels: u32,
+    pub m_faces: u32,
+    pub m_bytes_per_block: u32,
+    pub m_userdata0: u32,
+    pub m_userdata1: u32,
+    pub m_format: CrnFormat
+}
+
+impl CrnTextureInfo{
+    pub fn default() -> CrnTextureInfo{
+        return CrnTextureInfo {
+            m_struct_size: core::mem::size_of::<CrnTextureInfo>() as u32,
+			m_width: 0,
+			m_height: 0,
+			m_levels: 0,
+			m_faces: 0,
+			m_bytes_per_block: 0,
+			m_userdata0: 0,
+			m_userdata1: 0,
+			m_format: CrnFormat::CCrnfmtInvalid // Init as invalid?
+        }
+    }
+
+    pub fn crnd_get_texture_info(&mut self, p_data: &[u8], data_size: u32) -> bool{
+        if data_size < core::mem::size_of::<CrnHeader>() as u32 {
+            return false;
+        }
+
+        if self.m_struct_size != core::mem::size_of::<CrnTextureInfo>() as u32{
+            return false;
+        }
+
+        let mut p_header: CrnHeader = CrnHeader::default();

```

Comment:
`: CrnHeader` isn't needed here

```suggestion
        let mut p_header = CrnHeader::default();
```